### PR TITLE
updates needed for error improvements PR

### LINKED ISSUE #2374: Update docs for error improvements
## I'm submitting a...

```
[ ] Bug report
[x] Content update
[ ] Process update (build, deployment, ... )
```

## Type of bug / changes

- Documentation for https://github.com/cypress-io/cypress/pull/3930

- [x] Need link to https://on.cypress.io/file-opener-preference (clicked from 'learn more' in global settings where you change editor preferences
- [x] Update screenshots of reporter if necessary
- [ ] Update Command Log section to have section explaining errors
- [x] Need something to explain setting up source maps?

### PR DESCRIPTION
**Pull request by jennifer-shehane**
## TODO:

- [x] Do not merge until https://github.com/cypress-io/cypress/pull/3930 is merged for specific version

- close #2374 
- close #2421 (previous docs PR that had these changes)

Docs for error improvements including:

- New ‘Errors’ section in ‘Debugging’, ‘Test Runner’, and ‘Writing your first test’
- New ‘File Opener Preference’ section in IDE Integration doc
- updated screenshots of errors

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
@chrisbreiding I'm not sure if any changes have been made to the feature since we last wrote these that would affect the docs. Can you review?

File: source/_partial/errors_anatomy.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+1. **Error name**: This is the type of the error (e.g. AssertionError, CypressError)
+2. **Error message**: This generally tells you what went wrong. It can vary in length. Some are short like in the example, while some are long, and may tell you exactly how to fix the error. Some also contain a **Learn more** link that will take you to relevant Cypress documentation.
+3. **View stack trace**: Clicking this toggles the visibility of the stack trace.
+4. **Stack trace**: This is hidden by default, but clicking *View stack trace* will show it in full. Stack traces vary in length. Clicking on the blue file path will open the file in your [preferred file opener](https://on.cypress.io/IDE-integration#File-Opener-Preference).
```
Comment (chrisbreiding) [inline source/_partial/errors_anatomy.md]: Any reason these links are standard markdown links and aren't `{% url $}` tags?

File: source/guides/getting-started/writing-your-first-test.md [hunk 2]
```diff
@@ -86,6 +86,8 @@ describe('My First Test', function() {
 
 Once you save again, you'll see Cypress display the failing test in red since `true` does not equal `false`.
 
+Cypress also displays the stack trace and the code frame where the assertion failed (when available). You can click on the blue file link to open the file where the error occurred in your preferred application. To read more about the error's display, read about {% url "Debugging Errors" debugging#Errors %}.
```
Comment (chrisbreiding) [inline source/guides/getting-started/writing-your-first-test.md]: Should we link "your preferred application" to the file opener preference section?

File: source/guides/getting-started/writing-your-first-test.md [hunk 3]
```diff
@@ -189,7 +191,7 @@ Can you see what Cypress is doing under the hood? It's automatically waiting and
 {% imgTag /img/guides/first-test-failing-contains.png "Test failing to not find content 'hype'" %}
 
 {% note warning 'Error Messages' %}
-We've taken care at Cypress to write hundreds of custom error messages that attempt to clearly explain what went wrong. In this case Cypress **timed out retrying** to find the content: `hype` within the entire page.
+We've taken care at Cypress to write hundreds of custom error messages that attempt to clearly explain what went wrong. In this case Cypress **timed out retrying** to find the content: `hype` within the entire page. To read more about the error's display, read about {% url "Debugging Errors" debugging#Errors %}.
```
Comment (chrisbreiding) [inline source/guides/getting-started/writing-your-first-test.md]:
```suggestion
We've taken care at Cypress to write hundreds of custom error messages that attempt to clearly explain what went wrong. In this case, Cypress **timed out retrying** to find the content `hype` within the entire page. To read more about the error's display, read about {% url "Debugging Errors" debugging#Errors %}.
```

File: source/guides/tooling/IDE-integration.md [hunk 4]
```diff
@@ -2,6 +2,26 @@
...
+
+- File system (e.g. Finder on MacOS, File Explore on Windows)
+- An IDE located on your system
+- A specified application path
+
+{% note warning %}
+Cypress attempts to find available file editors on your system and display those as options. If your preferred editor is not list, you can specify the (full) path to it by selecting **Other**. Cypress will make every effort to open the file, *but it is not guaranteed to work with every application*.
+{% endnote %}
+
+After setting your file opener preference any files will automatically open in your selected application without prompting you to choose. If you want to change your selection, you can do so in the **Settings** tab of the Cypress Test Runner by clicking under **File Opener Preference**.
```
Comment (chrisbreiding) [inline source/guides/tooling/IDE-integration.md]:
```suggestion
After setting your file opener preference, any files will automatically open in your selected application without prompting you to choose. If you want to change your selection, you can do so in the **Settings** tab of the Cypress Test Runner by clicking under **File Opener Preference**.
```

File: source/_partial/errors_anatomy.md (hunk 1 above)
Comment (jennifer-shehane) [reply source/_partial/errors_anatomy.md]: 😅 Cause hexo is a mess that's why. url tags don't work in partials. https://github.com/cypress-io/cypress-documentation/issues/2422

File: source/guides/getting-started/writing-your-first-test.md (hunk 2 above)
Comment (jennifer-shehane) [reply source/guides/getting-started/writing-your-first-test.md]: Yes. updated 👍

**Pull request merged**